### DISCUSSION TITLE
Xiaomi Motion Sensor is always detecting movement

### DISCUSSION THREAD

**Discussion by jlassuncao** [merdok/homebridge-miot#569] [General]
Hi,
My two Xiaomi Motion Sensor (RTCGQ01LM) are always detecting movement....

![motion_sensor_HB](https://github.com/merdok/homebridge-miot/assets/149901273/58cd448b-c600-4d35-b069-25f728d36827)
![motion_sensor_HB_config](https://github.com/merdok/homebridge-miot/assets/149901273/72e17bcc-8ae7-4d3d-be75-5910bdb55f05)

I have window and door devices and work well. But the motions sensors are not working....

Can anyone help me?

**Comment by merdok** [comment]
I think it might be related to this issue https://github.com/merdok/homebridge-miot/pull/511
As currently we are not able to handle events of the device workarounds would need to be done which i am honestly not a fan off...

**Reply by jlassuncao** [reply]
Thanks, but i am running HB on a docker supported by Sonoff iHost hardware....
The python-miio not working so well on this device!

I have try see the .js files on the plugin patch (I dont understand anything of this, but...) and on the "lumi.sensor_motion.v2.js" I think that there is needded to define the movement and non-movement sent by ZigBee.
And on the "MotionSensorDevice.js" is necessary to add the motion-sensor:motion-detected var...

This is possible or make any sense?
ahahha, sory for the report

**Reply by jlassuncao** [reply]
I am saying this because, on one tentative I change on file "MotionSensroDevice.js" the property of "motion-sensor:motion-state" to "motion-sensor:motion-detected". And the sensor and the sensor stopped always reporting movement and started always reporting "no motion".

**Reply by merdok** [reply]
`motion-detected` is an event, and the events currently do not work hence you will not be able to get it running reliably...

**Reply by jlassuncao** [reply]
Ok, thanks anyway ;)
